statistics-reports.md

```diff
@@ -0,0 +1,202 @@
+# Displaying usage statistics
+[Back to the list of all defined endpoints](endpoints.md)
+
+## Statistics for a DSpaceObject
+**GET /api/statistics/usage-reports**
+
+As we don't have a use case to iterate over all the existing usage reports, the main endpoint is not implemented and a 405 error code is returned according to our [general error response codes](README.md#Error-codes).
+
+## Single statistic
+**GET /api/statistics/usage-reports/<:id>**
+
+This endpoint provides a specific statistic
+
+An example JSON response document to `/api/statistics/usage-reports/1911e8a4-6939-490c-b58b-a5d70f8d91fb_TopCountries`:
+```json
+{
+    "id": "1911e8a4-6939-490c-b58b-a5d70f8d91fb_TopCountries",
+    "type": "usage-report",
+    "report-type": "TotalVisits",
+    "points": [
```

Comment:
can we include the limit of points returned by the report as an attribute and as part of the report ID? i.e. `/api/statistics/usage-reports/1911e8a4-6939-490c-b58b-a5d70f8d91fb_TopCountries_0_10` would mean that a the first 10 points are returned?

can we add a link to extended report where the number of points can be changed?

Reply:
I've been wondering whether we should support changing the number of points, but only if we'd use a "usage-report" to represent the entire table, and we'd use e.g. a "usage-value" with paging to represent the rows in the table also with paging. I didn't include it because it's not supported in DSpace 6.

Reply:
I think we should have a fixed number of points for now (as that seems to be what was in DSpace 6, from what I can see).  So, for example, if I recall correctly, in DSpace 6 the "TopCountries" was only the top 10, and there was no way to page to the next ten.

In the future we could adjust or extend this to allow basic "pagination" to the next ten results, but I don't think that's necessary in this first iteration.